cohortextractor/measure.py

```diff
@@ -46,7 +46,7 @@ def calculate(self, data):
             data: a Pandas DataFrame
         """
         columns = _drop_duplicates([self.numerator, self.denominator, *self.group_by])
-        result = data[columns]
+        result = data[columns].copy()
```

Comment:
Is it worth a short comment here explaining that we're going to be mutating the data for low-number suppression?

Reply:
Yes. I think my lack of a comment here is a reflection of a failure of confidence on my part. I tried for a while, but I couldn't get a test to fail without the copy. So the only thing I _know_ the copy does is get rid of the warning.

(I left the test in because I think it's important that we _don't_ mutate the underlying data, but I'm embarrassed to admit that I've never seen it fail.)

I'll add the comment anyway. But any insight you might have into why I can't actually trigger the condition is welcome. My fear is that it's just that pandas (or numpy) moves in mysterious ways.

Reply:
If the comment said something like "We do X because if we don't Y happens for reasons we don't fully understand" it would join a long and inglorious lineage of similar comments scattered throughout our code :smile: (In fact, these are probably some of the most useful comments we have because they're the lines you're most tempted to remove.)